src/cascadia/TerminalApp/LeafPane.h

```diff
@@ -0,0 +1,121 @@
+                            const GUID& profile);
+        void ResizeContent(const winrt::Windows::Foundation::Size& /*newSize*/){};
+
+        TerminalApp::LeafPane Split(winrt::Microsoft::Terminal::Settings::Model::SplitState splitType,
+                                    const float splitSize,
+                                    const GUID& profile,
+                                    const winrt::Microsoft::Terminal::TerminalControl::TermControl& control);
+
+        float CalcSnappedDimension(const bool widthOrHeight, const float dimension) const;
+        void Shutdown();
+        void Close();
+        int GetLeafPaneCount() const noexcept;
+
+        uint16_t Id() noexcept;
+        void Id(uint16_t id) noexcept;
+
+        Borders2 Borders() noexcept;
+        void Borders(Borders2 borders) noexcept;
+
+        void Maximize(IPane paneToZoom);
+        void Restore(IPane paneToUnZoom);
+
+        winrt::Windows::Foundation::Size GetMinSize() const;
+
+        winrt::Windows::Foundation::IReference<winrt::Microsoft::Terminal::Settings::Model::SplitState> PreCalculateAutoSplit(const IPane target,
+                                                                                                                              const winrt::Windows::Foundation::Size parentSize) const;
+        winrt::Windows::Foundation::IReference<bool> PreCalculateCanSplit(const IPane target,
+                                                                          winrt::Microsoft::Terminal::Settings::Model::SplitState splitType,
+                                                                          const float splitSize,
+                                                                          const winrt::Windows::Foundation::Size availableSpace) const;
+
+        void UpdateBorderWithClosedNeighbor(TerminalApp::LeafPane closedNeighbor, const winrt::Microsoft::Terminal::Settings::Model::ResizeDirection& neighborDirection);
+
+        SnapSizeResult _CalcSnappedDimension(const bool widthOrHeight, const float dimension) const;
+
+        void BorderTappedHandler(winrt::Windows::Foundation::IInspectable const& sender,
+                                 winrt::Windows::UI::Xaml::Input::TappedRoutedEventArgs const& e);
+
+        void _UpdateBorders();
+
+        DECLARE_EVENT(Closed, _ClosedHandlers, winrt::delegate<LeafPane>);
+        DECLARE_EVENT(GotFocus, _GotFocusHandlers, winrt::delegate<LeafPane>);
+        DECLARE_EVENT(LostFocus, _LostFocusHandlers, winrt::delegate<LeafPane>);
+        DECLARE_EVENT(PaneRaiseVisualBell, _PaneRaiseVisualBellHandlers, winrt::delegate<LeafPane>);
+        TYPED_EVENT(PaneTypeChanged, IPane, IPane);
+
+    private:
+        winrt::Microsoft::Terminal::TerminalControl::TermControl _control{ nullptr };
+        GUID _profile;
+        bool _lastActive{ false };
+        uint16_t _id;
+        Borders2 _borders{ Borders2::None };
+        bool _zoomed{ false };
```

Comment:
Hmm... I wonder if there's any others we could simplify down to `GETSET_PROPERTY`. Might be worth checking.